docroot/modules/humsci/hs_revision_cleanup/src/RevisionCleanup.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace Drupal\hs_revision_cleanup;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * Service to delete old revisions on entity types.
+ *
+ * @package Drupal\hs_revision_cleanup
+ */
+class RevisionCleanup {
+
+  use StringTranslationTrait;
+
+  /**
+   * Database connection service.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $database;
+
+  /**
+   * Entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Database logger service.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * Config entity with cleanup settings.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected $config;
+
+  /**
+   * RevisionCleanup constructor.
+   *
+   * @param \Drupal\Core\Database\Connection $database
+   *   Database service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   Entity type manager service.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   Logger factory service.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   Config factory service.
+   */
+  public function __construct(Connection $database, EntityTypeManagerInterface $entity_type_manager, LoggerChannelFactoryInterface $logger_factory, ConfigFactoryInterface $config_factory) {
+    $this->database = $database;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->logger = $logger_factory->get('hs_revision_cleanup');
+    $this->config = $config_factory->get('hs_revision_cleanup.settings');
+  }
+
+  /**
+   * Delete all revisions for the configured entity types.
+   */
+  public function deleteRevisions() {
+    foreach ($this->config->get('cleanup') as $cleanup_entity) {
+      try {
+        // Subtract 1 from keep config to account for the current revision.
```

Comment:
```suggestion
        // Subtract 1 from keep config to account for the current revision because we filter out the current revision on the possible revisions query.
```